aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -504,17 +504,29 @@ Tensor to_dense_backward(const Tensor& grad, const Tensor& input_) {
     case kStrided:
       return grad.to_dense();
     case kSparse:
-      // Autograd operates on the coalesced assumption, i.e. no duplicate values.
-      return grad.sparse_mask(input_.coalesce());
+      if (at::globalContext().sparseSemantics()) {
+        return grad.to_sparse(input_.sparse_dim());
```

Comment:
I would agree that a no-op is best here. We don't need to try and enforce sparse/sparse correspondance between tensor and gradient.

Reply:
I agree. However, I think we are not there yet to enable this for sparse tensors in general.

For instance, when returning `grad`, this will lead to
```
test/test_autograd.py::TestAutograd::test_gradcheck_backward_mul_by_grad_output FAILED
...
_________________________________________________________________________
RuntimeError: hook 'hook' has changed the type of value (was torch.DoubleTensor got torch.sparse.DoubleTensor)
```

Reply:
The test you reference should probably be removed. 

Also the checks internal to gradcheck that enforce layout equality between inputs and their gradients on the backward path should go as well. 

In general backward implementations can make no assumption about the layout of the incoming grad and should not do anything to artificially impose a layout on the gradients coming out.

Reply:
> The test you reference should probably be removed.

Possibly. I don't know the extent of this issue yet to decide if this suggestion should be implemented in this PR or in a follow-up, especially when the feature requires "plenty of discussions".